internal/publisher/core_test.go

```diff
@@ -0,0 +1,82 @@
+package publisher
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	brokerstoremock "github.com/razorpay/metro/internal/brokerstore/mocks"
+	"github.com/razorpay/metro/pkg/messagebroker"
+	mockMB "github.com/razorpay/metro/pkg/messagebroker/mocks"
+	metrov1 "github.com/razorpay/metro/rpc/proto/v1"
+	"github.com/rs/xid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCore_Publish(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	mockBrokerStore := brokerstoremock.NewMockIBrokerStore(ctrl)
+	producer := mockMB.NewMockProducer(ctrl)
+	publisherCore := NewCore(mockBrokerStore)
+	req := getDummyPublishRequest()
+	msgID := xid.New().String()
```

Comment:
@sakshi-goyal-razorpay why are we using xid instead of UUID here?

Reply:
This is what we use while assigning message id
https://github.com/razorpay/metro/blob/master/pkg/messagebroker/kafka_utils.go#L220